tests/pytests/functional/modules/state/test_state.py

```diff
@@ -784,6 +785,7 @@ def test_retry_option_success_parallel(state, state_tree, tmp_path):
             assert "Attempt 2" not in state_return.comment
 
 
+@pytest.mark.xfail(reason="This test is flaky")
```

Comment:
Why not mark it as flaky?

Reply:
I tried `flaky(max_runs=10)`, but even with that we were still getting failures in some OSes, where duration is lower than the expected 4 seconds.

So, I set this to "xfail" until this test can be repaired.

Reply:
I'm wary of just turning off some tests because they don't work in your pipelines but are working in the salt pipelines.  What is the difference in the CICD pipelines?  At the very least we could add a marker that turns them off if a certain flag is present so that you can turn them off for your pipelines, but they can still run in the salt pipelines.

Reply:
Thanks for your comment @Akm0d !

I understand your point and I would agree on that. The thing here is that I think the test is not robust enough, making it to fail on certain environments (i.a. with different performance).

As you see in https://github.com/saltstack/salt/pull/66630/files#diff-88872a466f3936f773057e2bcf5e3c109242bac81cf3d89b8be00777982a5712R828 the test is expecting a duraction greater than 4 seconds, while in our CI this takes usually less than 4 seconds (~ 3.6-3.9 seconds) and then it fails.

If I look at the test, I see here https://github.com/saltstack/salt/pull/66630/files#diff-88872a466f3936f773057e2bcf5e3c109242bac81cf3d89b8be00777982a5712R828 that we only sleep for 2 seconds, but we expect a duraction of > 4 seconds. Maybe that is not high enough sleeping time to ensure the duration that big.

Would you be ok to increase the `time.sleep` to 3-4 seconds at least, then we would ensure duration is at least that high? Alternatively I could change the expected duration to `> 2 seconds`, that should also make the test more reliable.